src/index.html

```diff
@@ -316,6 +352,27 @@ <h1 class="text-white">How can we help?</h1>
         changeHeroTextAndImage(true);
         interval = setInterval(changeHeroTextAndImage, 5000, true);
       });
+      //Add desktop header on scroll or resize of the page
+      window.onscroll = function () {
+        evalHeaderOnChange(document);
+      };
+      window.onresize = function () {
+        evalHeaderOnChange(document);
+      };
+      let showHeader = false;
+      function evalHeaderOnChange(doc) {
+        var header = doc.getElementById("header");
+        var subHero = doc.getElementById("subHero");
+        // 64 px corresponds to h-16, 640 px corresponds to sm breakpoint
+        var sticky = showHeader ? subHero.offsetTop - 64 : subHero.offsetTop;
+        if (window.pageYOffset > sticky && window.innerWidth >= 640) {
+          header.classList.remove("sm:hidden");
+          showHeader = true;
+        } else {
+          header.classList.add("sm:hidden");
+          showHeader = false;
+        }
+      }
```

Comment:
- Line 367 is guarding against a race condition that causes the header on desktop to flash if you scroll very slowly past the breakpoint at the bottom of the hero component. The YOffset of the `subHero` section changes when the header is added, which sets off the race condition. So the boolean variable is meant to remedy that.

all the other suggestions make sense

Reply:
Ah, I see, thanks for clarifying. I can see where there could be a race condition here and have been trying to replicate this, but haven't been able to. To better understand why, I added a log statement to see the relationship between the `subHeroOffset` and `pageYOffset` (see screenshot) and from this you can see where the jump from "no header" to "header" is. When the header appears, both values jump 64 pixels together, so it seems like there should be no way for the page to ever be at an offset less than the `subHero` component.

That being said, are you still seeing that behavior with these changes? If not, can you do the same thing, add a log statement to print out both subHeroOffset and pageYOffset? I think it would be very helpful here for my own learning, too, to isolate what causes this difference in behavior.

Reply:
I'm actually only seeing this on Firefox, on chrome it works as you demo'd without the guard.
To recreate it (in case this is not how you were already trying to recreate it) change the following line
`var sticky = showHeader ? subHero.offsetTop - 64 : subHero.offsetTop;`
to 
`var sticky = subHero.offsetTop;`
Here is the behavior without the guard, and you can see adding the header changes to subHero offset.
![demo-pageoffset-unguarded](https://user-images.githubusercontent.com/16727618/205312687-d6005f3c-4427-46ce-bdb1-f77ee088e76d.png)

Reply:
Interesting! Thanks for providing this extra context. Firefox can be so tricky! You're right that we need a safeguard. Out of the interest of time, let's keep it as a mutable global variable rather than trying to rethink this strategy and just be careful with it. So I think something like this should work:

**index.html**
```
let showHeader = false;
// Add desktop header on scroll or resize of the page
window.onscroll = function () {
        showHeader = evalHeaderOnChange(showHeader);
};
```

**index.js**
```
function evalHeaderOnChange(showHeader) {
  const header = document.getElementById("header");
  const subHero = document.getElementById("subHero");
  // 64 px corresponds to h-16, 640 px corresponds to sm breakpoint
  const sticky = showHeader ? subHero.offsetTop - 64 : subHero.offsetTop;
  if (window.pageYOffset > sticky && window.innerWidth >= 640) {
    header.classList.remove("sm:hidden");
    showHeader = true;
  } else {
    header.classList.add("sm:hidden");
    showHeader = false;
  }
  return showHeader;
}
```

Reply:
In theory you and I will be the only ones referencing this code, but just in case, I would also suggest adding a bit more of a detailed comment to why `sticky` is conditional. I think renaming `sticky` to something like `headerPosition` may make this a bit clearer, as well.

Reply:
sounds good. This [commit](https://github.com/azavea/raster-vision-website2/pull/27/commits/3e417664b8a98a8d35d63c58ad5e2e453436a2e4) I added a few moments ago applies all the changes in the js as you wrote in the comment above. I will add one more commit with a comment in the javascript as to why the boolean global variable is necessary.